server/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -348,9 +481,19 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
             if (fieldType().omitNorms()) {
                 createFieldNamesField(context, fields);
             }
+            if (prefixFieldMapper != null) {
+                prefixFieldMapper.addField(value, fields);
+            }
         }
     }
 
+    @Override
+    public Iterator<Mapper> iterator() {
+        if (prefixFieldMapper == null)
+            return super.iterator();
+        return Iterators.concat(multiFields.iterator(), Collections.singleton(prefixFieldMapper).iterator());
```

Comment:
maybe replace `multiFields.iterator()` with `super.iterator()` to be more future-proof in case the super impl is updated in the future?